### PR TITLE
Blog: Quarkus 2.x platform and Quarkiverse support in the dev tools

### PR DESCRIPTION
**Pull request by aloubyansky**
An announce of the availability of the new Quarkus 2.x platform model and integration of the Quarkiverse extensions into the dev tools.

### CODE REVIEW

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+the higher the risk for its users of running into incompatiblity issues including libraries that aren't managed by the BOM into their applications.
+
+The `io.quarkus:quarkus-universe-bom`, as its artifactId implies, is certainly a large and diverse BOM, including dependencies of Camel, Google Cloud Services, Kogito, OptaPlanner
+and other Quarkus platform members. Importing `io.quarkus:quarkus-universe-bom` will enforce ~3600 dependency version constraints on an application. Besides the fact that it's practically
+always way more than necessary, it may actually cause a serious compatibility issue that could have been avoided if the Quarkus platform BOM was not trying to manage the "universe".
+
+E.g. including a library that depends on a different version of the `commons-beanutils:commons-beanutils` than Camel Quarkus into an application that actually does not include any
+Camel Quarkus extension may become problematic, because the `io.quarkus:quarkus-universe-bom` will enforce the vesion required by Camel Quarkus.
```
Comment (gastaldi) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:22]:
```suggestion
Camel Quarkus extension may become problematic, because the `io.quarkus:quarkus-universe-bom` will enforce the version required by Camel Quarkus.
```

**Review by maxandersen — COMMENTED** [review]
Content is good but I think it goes too deep to quickly and have the biggest user impact at end rather than beginning. The bom part is relevant but imo it's truly secondary to the main problem we are solving - enabling browsing and using non platform extensions (like those in quarkiverse). 

We must Remember most users doesn't even realize that the universe bom actually is just one part of a platform with metadata driving rest of tools and it's this part we are adjusting/expanding. Now is a good time to give users a overview on how it works and what it means for them as quarkus users and extension writers. 

Doesn't have to be super long just enough to connect with what readers will get most value from. Then go into details. Ie. Cli examples are great and I would put those upfront when going deeper.
 from there explain the advantage of the multi member poms and the less constrained surface.

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    </dependencies>
+  </dependencyManagement>
+```
+
+which is an equivalent of the `io.quarkus:quarkus-bom` that does not include any of the other platform members, such as Camel, Kogito, OptaPlanner, etc.
+
+However, if we did want to create a new project including a Kogito extension, it could be done like this:
+
+```shell
+quarkus create kogito-quarkus-rules --registry-client
```
Comment (gsmet) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: I think I'm a bit confused by the CLI, `kogito-quarkus-rules` is not the name of the project but a dependency?

Comment (aloubyansky) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: yes, it's an extension artifactId.

**Review by gsmet — APPROVED** [review]

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc (hunk 2 above)
Comment (gastaldi) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: For the record, I made the same assumption as @gsmet about expecting the project name in the CLi args

Comment (aloubyansky) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: right, it's a bit confusing

Comment (aloubyansky) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: @ebullient just fyi

Comment (maxandersen) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: FYI, maven plugin been able to do this since day one too afaik. 

Just that our examples always verbose.

Comment (maxandersen) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: btw. i would make the registry- client be first as the list of extension to look up is the last arguments.

```suggestion
quarkus create --registry-client kogito-quarkus-rules 
```

also just noticed the cli does not seem to lookup partial names as was done in past. ill open separate issue for that

Comment (aloubyansky) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]: > FYI, maven plugin been able to do this since day one too afaik. 
> 
> Just that our examples always verbose. 

I don't think so. It needs -Dextensions=

Comment (ebullient) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc]: This is what the cli has always done (re: listing extensions).
Is it more natural to choose the artifactId (only) without an option, and require an option for additional extensions?

Comment (gastaldi) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc]: @ebullient yes, I think it's more natural to choose the artifactId and require an option for additional extensions

Like: `quarkus create app foo --with-extensions prettytime quartz`

Comment (aloubyansky) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc]: yes, but i'm pretty sure people will want to specify the groupId instead of using the default one. Should we go for `quarkus create <GAV> --extensions xxx`?

Comment (ebullient) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc]: That would be a big change to make now (especially since this is how the CLI create path always worked.. free typing of extensions against default group/artifact/version). I mean.. the CLI wasn't advertised much, so you could argue that not many know it worked that way...

What you've shown is half-positional. `--with-extensions=one,two,three` or `-x one -x two -x three` would be more typical.

Comment (ebullient) [reply _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc]: See quarkusio/quarkus#18368 for updated create syntax

**Comment by aloubyansky** [comment]
Erin did a great job changing the command syntax as we discussed but it didn't get backported to 2.0.1.Final. So, I leaving the commands as they are and they will look different in 2.1.0.Final?

**Comment by ebullient** [comment]
Sorry. I left for PTO at an awkward time and must have missed a tag. Can just add a note.

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 3]
```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title: 'The New Quarkus 2.x Platform Model and Ecosystem Support in the Dev Tools'
+date: 2021-07-01
+tags: release
+synopsis: 'The Quarkus 2.0.0.Final release introduces the new Quarkus platform model and the dev tools that make it easy to discover and work with the Quarkus extension ecosystem.'
+author: aloubyansky
+---
+
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:9]:
```suggestion
= Platforms and Streams: a new way to discover Quarkus extensions

As you may have seen from recent Quarkus Insights episodes (#48, #55 and #57, specifically), we've made a number of changes to our tools in the 2.x stream to improve the developer experience.  When 2.0 was released, a few sharp-eyed Quarkus client users spotted some new options, and we're excited to explain what those are all about with the release of Quarkus 2.1.

**TL;DR:** We have a whole new set of more granular BOMs that you can use instead of the big `quarkus-universe-bom`. We have deployed a registry service that aligns and manages platform compatibility across these more granular BOMs, and our tools (maven, gradle, and client) know how to work with this registry to simplify the discovery of extensions that work with your project.

As of the 2.1.0.Final release, the Quarkus client uses this registry by default to resolve the Quarkus platform used by your project.

[source, bash]
----
# The client will create a project with the latest/recommended quarkus release
$ quarkus create
# You can use the --stream parameter to narrow to the latest release 
$ quarkus create -S 2.0
$ quarkus create -S 2.1
----

```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+= The Quarkus 2.x Platform Model
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:10]:
```suggestion
== The Evolution of the Platform Model
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+the higher the risk for its users of running into incompatiblity issues including libraries that aren't managed by the BOM into their applications.
+
+The `io.quarkus:quarkus-universe-bom`, as its artifactId implies, is certainly a large and diverse BOM, including dependencies of Camel, Google Cloud Services, Kogito, OptaPlanner
+and other Quarkus platform members. Importing `io.quarkus:quarkus-universe-bom` will enforce ~3600 dependency version constraints on an application. Besides the fact that it's practically
+always way more than necessary, it may actually cause a serious compatibility issue that could have been avoided if the Quarkus platform BOM was not trying to manage the "universe".
+
+E.g. including a library that depends on a different version of the `commons-beanutils:commons-beanutils` than Camel Quarkus into an application that actually does not include any
+Camel Quarkus extension may become problematic, because the `io.quarkus:quarkus-universe-bom` will enforce the version required by Camel Quarkus.
+
+The Quarkus 2.0.0.Final release, in addition to the monolithic `io.quarkus:quarkus-universe-bom`, also includes the platform member-specific BOMs, e.g.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:24]:
```suggestion
As of Quarkus 2.0.0.Final, in addition to the monolithic `io.quarkus:quarkus-universe-bom`, we also define a number of platform-member-specific BOMs, e.g.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+* `io.quarkus.platform:quarkus-bom:2.0.0.Final` - an equivalent of the `io.quarkus:quarkus-bom`;
+* `io.quarkus.platform:quarkus-camel-bom:2.0.0.Final` - manages only the Camel Quarkus-related extensions and their required dependencies;
+* `io.quarkus.platform:quarkus-kogito-bom:2.0.0.Final` - manages only the Kogito Quarkus-related extensions and their required dependencies;
+* etc
+
+Having member-specific BOMs, applications now can import only the BOMs that are relevant to them and avoid enforcing dependency constraints from the rest of the "universe".
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:31]:
```suggestion
Now applications only need to import relevant the BOMs, which avoids enforcing dependency constraints from the rest of the "universe".
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+NOTE: Given that every member-specific BOM is actually a fragment of the `io.quarkus:quarkus-universe-bom`, member BOMs can be imported in any order w/o creating a conflict.
+
+== Dev Tools Support
+
+The dev tools, such as the Quarkus CLI, Maven or Gradle plugins, can be used to create and manage a Quarkus application project using the new platform model.
+
+NOTE: The `io.quarkus:quarkus-universe-bom` is still the default BOM in Quarkus 2.0.0.Final but in one of the subsequent Quarkus 2.x releases it will be deprecated
+in favor of the member-specific BOMs.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:40]:
```suggestion
member-specific BOMs instead.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+Having member-specific BOMs, applications now can import only the BOMs that are relevant to them and avoid enforcing dependency constraints from the rest of the "universe".
+
+NOTE: Given that every member-specific BOM is actually a fragment of the `io.quarkus:quarkus-universe-bom`, member BOMs can be imported in any order w/o creating a conflict.
+
+== Dev Tools Support
+
+The dev tools, such as the Quarkus CLI, Maven or Gradle plugins, can be used to create and manage a Quarkus application project using the new platform model.
+
+NOTE: The `io.quarkus:quarkus-universe-bom` is still the default BOM in Quarkus 2.0.0.Final but in one of the subsequent Quarkus 2.x releases it will be deprecated
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:39]:
```suggestion
NOTE: The `io.quarkus:quarkus-universe-bom` was still the default BOM in Quarkus 2.0.0.Final. As of Quarkus 2.1.0.Final, we recommend using the more granular, 
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+in favor of the member-specific BOMs.
+
+The easiest way to play with it would be using the new https://quarkus.io/guides/cli-tooling[Quarkus CLI], which can be installed using e.g.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:42]:
```suggestion
The easiest way to explore this new platform resolution capability is using the new https://quarkus.io/guides/cli-tooling[Quarkus CLI], which can be installed using e.g.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+```shell
+curl -Ls https://sh.jbang.dev | bash -s - app install --fresh --force quarkus@quarkusio
+```
+
+Once installed you can create a project by executing
+
+```shell
+quarkus create app --registry-client
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:51]:
```suggestion
quarkus create app
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+NOTE: Without the `--registry-client` argument the created project will be importing the default `io.quarkus:quarkus-universe-bom:2.0.0.Final`.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:54]:
```suggestion
NOTE: For earlier versions of the CLI (pre-2.1.0.Final), you need to specify the `--registry-client` option, or the created project will use the `io.quarkus:quarkus-universe-bom` associated with the client's version.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+
+The generated `pom.xml` will be importing the following BOM:
+```xml
+    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>2.0.0.Final</quarkus.platform.version>
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:60]:
```suggestion
    <quarkus.platform.version>2.1.0.Final</quarkus.platform.version>
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+```
+
+which is an equivalent of the `io.quarkus:quarkus-bom` that does not include any of the other platform members, such as Camel, Kogito, OptaPlanner, etc.
+
+However, if we did want to create a new project including a Kogito extension, it could be done like this:
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:78]:
```suggestion
Let's create another project that does include a Kogito extension:
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+
+```shell
+quarkus create --registry-client kogito-quarkus-rules
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:81]:
```suggestion
quarkus create -x kogito-quarkus-rules kogito-app
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+
+which is an equivalent of the `io.quarkus:quarkus-bom` that does not include any of the other platform members, such as Camel, Kogito, OptaPlanner, etc.
+
+However, if we did want to create a new project including a Kogito extension, it could be done like this:
+
+```shell
+quarkus create --registry-client kogito-quarkus-rules 
+```
+
+And the generated project would now import two BOMs (i.e. two fragments of the `io.quarkus:quarkus-universe-bom` that are relevant to the project):
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:84]:
```suggestion
The new project imports two BOMs (i.e. two fragments of the `io.quarkus:quarkus-universe-bom` that are relevant to the project):
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+
+```xml
+    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>2.0.0.Final</quarkus.platform.version>
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:89]:
```suggestion
    <quarkus.platform.version>2.1.0.Final</quarkus.platform.version>
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-quarkus-rules</artifactId>
+    </dependency>
+```
+
+= Wider Quarkus Ecosystem Support in the Dev Tools
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:117]:
```suggestion
== Wider Quarkus Ecosystem Support in the Dev Tools
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+
+https://quarkus.io/guides/platform[Quarkus platform] does not represent the whole Quarkus extension ecosystem, aka Quarkiverse, but rather a set of extensions that target
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:119]:
```suggestion
The https://quarkus.io/guides/platform[Quarkus platform] does not represent the whole Quarkus extension ecosystem, aka Quarkiverse, but rather a set of extensions that target
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+the primary use-cases of Quarkus as a development stack. Which means that there are still plenty of Quarkus extensions that aren't present in a Quarkus platform (BOM),
+e.g. most of the extensions hosted on the https://github.com/quarkiverse[Quarkiverse Hub]. These non-platform extensions could still be added
+to Quarkus applications as usual application dependencies. The Quarkus 2.0.0.Final dev tools make it very easy, e.g.
+
+```shell
+quarkus create prettytime --registry-client
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:125]:
```suggestion
quarkus create -x prettytime test-time-app
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+      <artifactId>kogito-quarkus-rules</artifactId>
+    </dependency>
+```
+
+= Wider Quarkus Ecosystem Support in the Dev Tools
+
+https://quarkus.io/guides/platform[Quarkus platform] does not represent the whole Quarkus extension ecosystem, aka Quarkiverse, but rather a set of extensions that target
+the primary use-cases of Quarkus as a development stack. Which means that there are still plenty of Quarkus extensions that aren't present in a Quarkus platform (BOM),
+e.g. most of the extensions hosted on the https://github.com/quarkiverse[Quarkiverse Hub]. These non-platform extensions could still be added
+to Quarkus applications as usual application dependencies. The Quarkus 2.0.0.Final dev tools make it very easy, e.g.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:122]:
```suggestion
to Quarkus applications as usual application dependencies. The Quarkus 2.1.0.Final dev tools make it very easy, e.g.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+```shell
+quarkus create prettytime --registry-client
+```
+
+will create a new Quarkus project with a non-platform https://github.com/quarkiverse/quarkus-prettytime[Quarkus Pretty Time extension].
+
+```xml
+    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>2.0.0.Final</quarkus.platform.version>
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:133]:
```suggestion
    <quarkus.platform.version>2.1.0.Final</quarkus.platform.version>
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+    <dependency>
+      <groupId>io.quarkiverse.prettytime</groupId>
+      <artifactId>quarkus-prettytime</artifactId>
+      <version>0.1.0</version>
+    </dependency>
+```
+
+== Registry Client Support in the Maven and Gradle Plugins
+
+The Quarkus Maven and Gradle plugins could still be used to manage Quarkus projects. The equivalent of the CLI's `--registry-client` argument using the plugins would be
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:157]:
```suggestion
The Quarkus Maven and Gradle plugins can still be used to manage Quarkus projects.
```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+`-DquarkusRegistryClient` or environment variable `QUARKUS_REGISTRY_CLIENT` set to `true`, e.g.
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:158]:
```suggestion

```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 24]
```diff
@@ -0,0 +1,165 @@
...
+
+```shell
+mvn io.quarkus:quarkus-maven-plugin:2.0.0.Final:create \
+    -Dextensions=kogito-quarkus-rules,prettytime \
+    -DprojectGroupId=org.acme -DprojectArtifactId=quarkus-app -DprojectVersion=1.0 \
+    -DquarkusRegistryClient
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:164]:
```suggestion

```

File: _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 25]
```diff
@@ -0,0 +1,165 @@
...
+    </dependency>
+```
+
+== Registry Client Support in the Maven and Gradle Plugins
+
+The Quarkus Maven and Gradle plugins could still be used to manage Quarkus projects. The equivalent of the CLI's `--registry-client` argument using the plugins would be
+`-DquarkusRegistryClient` or environment variable `QUARKUS_REGISTRY_CLIENT` set to `true`, e.g.
+
+```shell
+mvn io.quarkus:quarkus-maven-plugin:2.0.0.Final:create \
```
Comment (ebullient) [inline _posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc:161]:
```suggestion
mvn io.quarkus:quarkus-maven-plugin:2.1.0.Final:create \
```

**Review by ebullient — CHANGES_REQUESTED** [review]

**Review by ebullient — APPROVED** [review]

**Comment by aloubyansky** [comment]
@gsmet I've changed the date to 30.07.2021 (after the platform release). Is it possible to have multiple authors?

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 26]
```diff
@@ -0,0 +1,179 @@
...
+date: 2021-07-30
+tags: release
+synopsis: 'The platform model and supporting tools available in Quarkus 2.1.0.Final make it easy to discover and work with the Quarkus extension ecosystem.'
+author: aloubyansky
+---
+= Platforms and Streams: a new way to discover Quarkus extensions
+
+As you may have seen from recent Quarkus Insights episodes (#48, #55 and #57, specifically), we've made a number of changes to our tools in the 2.x stream to improve the developer experience.  When 2.0 was released, a few sharp-eyed Quarkus client users spotted some new options and we're excited to explain what those are all about with the release of Quarkus 2.1.
+
+**TL;DR:** We have a whole new set of more granular BOMs that you can use instead of the big `quarkus-universe-bom`. We have deployed a registry service that aligns and manages platform compatibility across these more granular BOMs and our tools (maven, gradle and client) know how to work with this registry to simplify the discovery of extensions that work with your project.
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:13]:
```suggestion
**TL;DR:** We have a whole new set of more granular BOMs that you can use instead of the big `quarkus-universe-bom`. We have deployed a registry service that aligns and manages platform compatibility across these more granular BOMs and our tools (maven, gradle and CLI) know how to work with this registry to simplify the discovery of extensions that work with your project.
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 27]
```diff
@@ -0,0 +1,179 @@
...
+
+As of the 2.1.0.Final release, the Quarkus client uses this registry by default to resolve the Quarkus platform used by your project.
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:15]:
```suggestion
As of the 2.1.0.Final release, the Quarkus CLI uses this registry to resolve the Quarkus platform used by your project.
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 28]
```diff
@@ -0,0 +1,179 @@
...
+$ quarkus create -S 2.0
+$ quarkus create -S 2.1
+----
+
+== The Evolution of the Platform Model
+
+The Quarkus 1.x platform was based on a single BOM - `io.quarkus:quarkus-universe-bom`, that would be imported by Quarkus applications. The universe BOM would include all the Quarkus
+platform extensions plus their dependencies. BOMs are meant to make it easy to align common dependencies on versions that work for all the libraries and frameworks used in an application.
+However, badly composed BOMs may actually make it very difficult or practically impossible to achieve that. Generally speaking, the wider the scope of the dependencies a BOM manages,
+the higher the risk for its users of running into incompatiblity issues including libraries that aren't managed by the BOM into their applications.
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:31]:
```suggestion
the higher the risk of running into incompatiblity issues when users include libraries that aren't managed by the BOM into their applications.
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 29]
```diff
@@ -0,0 +1,179 @@
...
+
+The `io.quarkus:quarkus-universe-bom`, as its artifactId implies, is certainly a large and diverse BOM, including dependencies of Camel, Google Cloud Services, Kogito, OptaPlanner
+and other Quarkus platform members. Importing `io.quarkus:quarkus-universe-bom` will enforce ~3600 dependency version constraints on an application. Besides the fact that it's practically
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:34]:
```suggestion
and other Quarkus platform members. Importing `io.quarkus:quarkus-universe-bom` will enforce ~3600 dependency version constraints on an application. 

Besides the fact that it's practically
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 30]
```diff
@@ -0,0 +1,179 @@
...
+always way more than necessary, it may actually cause a serious compatibility issue that could have been avoided if the Quarkus platform BOM was not trying to manage the "universe".
+
+E.g. including a library that depends on a different version of the `commons-beanutils:commons-beanutils` than Camel Quarkus into an application that actually does not include any
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:37]:
```suggestion
For example, including a library that depends on a different version of the `commons-beanutils:commons-beanutils` than Camel Quarkus into an application that actually does not include any
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 31]
```diff
@@ -0,0 +1,179 @@
...
+platform extensions plus their dependencies. BOMs are meant to make it easy to align common dependencies on versions that work for all the libraries and frameworks used in an application.
+However, badly composed BOMs may actually make it very difficult or practically impossible to achieve that. Generally speaking, the wider the scope of the dependencies a BOM manages,
+the higher the risk for its users of running into incompatiblity issues including libraries that aren't managed by the BOM into their applications.
+
+The `io.quarkus:quarkus-universe-bom`, as its artifactId implies, is certainly a large and diverse BOM, including dependencies of Camel, Google Cloud Services, Kogito, OptaPlanner
+and other Quarkus platform members. Importing `io.quarkus:quarkus-universe-bom` will enforce ~3600 dependency version constraints on an application. Besides the fact that it's practically
+always way more than necessary, it may actually cause a serious compatibility issue that could have been avoided if the Quarkus platform BOM was not trying to manage the "universe".
+
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:36]:
```suggestion
```

File: _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc [hunk 32]
```diff
@@ -0,0 +1,179 @@
...
+As of Quarkus 2.0.0.Final, in addition to the monolithic `io.quarkus:quarkus-universe-bom`, we also define a number of platform-member-specific BOMs, e.g.
+
+* `io.quarkus.platform:quarkus-bom:2.1.0.Final` - an equivalent of the `io.quarkus:quarkus-bom`;
+* `io.quarkus.platform:quarkus-camel-bom:2.1.0.Final` - manages only the Camel Quarkus-related extensions and their required dependencies;
+* `io.quarkus.platform:quarkus-kogito-bom:2.1.0.Final` - manages only the Kogito Quarkus-related extensions and their required dependencies;
+* etc
+
+Now applications need to import only the relevant BOMs, which avoids enforcing dependency constraints from the rest of the "universe".
+
+NOTE: Given that every member-specific BOM is actually a fragment of the `io.quarkus:quarkus-universe-bom`, member BOMs can be imported in any order w/o creating a conflict.
```
Comment (ebullient) [inline _posts/2021-07-30-quarkus-2x-platform-quarkiverse-registry.adoc:49]:
```suggestion
TIP: Given that every member-specific BOM is actually a fragment of the `io.quarkus:quarkus-universe-bom`, member BOMs can be imported in any order w/o creating a conflict.
```

**Review by ebullient — COMMENTED** [review]
Some additional suggestions on a re-read. I feel like another pass could be made on the bottom-half still.

**Comment by aloubyansky** [comment]
Not sure whether it'll work but i changed the author to `author: aloubyansky,ebullient`
